### ISSUE TITLE
 Training questions

### ISSUE THREAD

**Issue by hazemati** [AlexeyAB/darknet#1707]
I had one more question, so on one hand you've been saying that we need 2000 images to train, on the other hand, you've been saying that that training is designed to go on forever and I need to stop it. 1) Could you please explain this 2) how long should I expect it to take to train 2000 images. 3) Does the resoluiton of the images that I'm training on matter?

**Comment by AlexeyAB** [comment]
1. You should have at least 2000 images per class, and you should train at least 2000 iterations per class. So if you have 5 classes, then you should have at least 10 000 images, and set `max_batches=10000` and `steps=8000,9000` But in general: the more images and iterations - the better accuracy.

2. It will take `2000/batch = 2000/64 = 31` iterations to train 2000 images. But you should train at least 2000 iterations - so data augmentation algorithm will modify these images and create 128 000 images automatically in memory.

3. Usually resolution of images doesn't matter. But if you want to get the maximum accuracy, then you should follow general rule: https://github.com/AlexeyAB/darknet/issues/1668#issuecomment-424868065

https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

> General rule - your training dataset should include such a set of relative sizes of objects that you want to detect:
> 
> `train_network_width * train_obj_width / train_image_width ~= detection_network_width * detection_obj_width / detection_image_width`

**Comment by AlexeyAB** [comment]
> I've tried training on 1 image, and verified that I can train that 1 image for 30 iterations (with data augmentation)

Did you check it, i.e. did you train a model with 1 image for 30 iterations, and then can you detect anything? I can't detect anyting after 30 iterations.

----

In general:

Required images >= `classes*2000`

Required iterations >= `max(4000, number_of_images, classes*2000)`

**Comment by hazemati** [comment]
I totally agree with you it's not the best test because I wouldn't get a working model. But  my end goal is to approximate how long it takes to train those 2000 images. Right now, it takes me 12 minutes to get through training with 1 image... days to train all 2000 images. My point is, I'm doing something wrong...

**Comment by AlexeyAB** [comment]
Just train about 4000 iterations for 1 image with `batch=64 subdivisions=32` in your cfg-file.
